### DISCUSSION TITLE
Same flow with different contexts and Labels

### DISCUSSION THREAD

**Discussion by fosk06** [PrefectHQ/prefect#4892] [General]
Hello prefect community.
I have a flow that i need to run on "production" environment and "staging" env.
Is it possible to run the SAME flow but changing the context and labels ( i have one agent for staging and one for production). 
Right now i could'nt find another solution than make 2 differents flows .

**Comment by fosk06** [comment]
`run_config_dev = KubernetesRun(
        image=os.getenv('PREFECT_DOCKER_IMAGE'),
        labels = ['dev'],
        image_pull_policy="Always"
    )
    with Flow("ge-build-data-docs", run_config=run_config_dev,storage=GCS(bucket=os.getenv('PREFECT_BUCKET'))) as flow:
        build_data_doc('GE_WRITER_SA_DEVELOPMENT') # run in dev

    # schedule = Schedule(clocks=[CronClock("0 0 * * *")])
    # flow.schedule(schedule)
    flow.register(project_name="great-expectations")`

**Reply by fosk06** [reply]
I would like tu use the same Flow twice but changing the label ["dev"] to ["prod"] and setting the "GE_WRITER_SA_DEVELOPMENT" in the context instead of hard coding it.

**Comment by zanieb** [comment]
Hey @fosk06 -- you could leave these unset on the `Flow` and just set them when each flow run is created.

**Reply by fosk06** [reply]
i can do it on the UI ( prefect cloud for sure ) but can i dot it progamatically ? i would like to setup all of this in the code.

**Reply by zanieb** [reply]
Yep, you can do this with the Prefect client. See https://docs.prefect.io/orchestration/flow-runs/creation.html#python-client

`Client.create_flow_run` accepts `context` and `labels`
